aspnetcore/mvc/advanced/custom-formatters/Sample/Formatters/VcardOutputFormatter.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Text;
+using CustomFormatterDemo.Models;
+using Microsoft.Net.Http.Headers;
+using System.Reflection;
+using Microsoft.Extensions.Logging;
+
+namespace CustomFormatterDemo.Formatters
+{
+    #region classdef
+    public class VcardOutputFormatter : TextOutputFormatter
+    #endregion
+    {
+        #region ctor
+        public VcardOutputFormatter()
+        {
+            SupportedMediaTypes.Add(MediaTypeHeaderValue.Parse("text/vcard"));
+
+            SupportedEncodings.Add(Encoding.UTF8);
+            SupportedEncodings.Add(Encoding.Unicode);
+        }
+        #endregion
+
+        #region canwritetype
+        protected override bool CanWriteType(Type type)
+        {
+            if (typeof(Contact).IsAssignableFrom(type) || typeof(IEnumerable<Contact>).IsAssignableFrom(type))
```

Comment:
newline here so we don't get a horz scroll bar with the largest browser window with no left or right TOC.